sfr_eff.c

```diff
@@ -449,10 +450,12 @@ static int get_sfr_condition(int i) {
         flag = 1;		/* only normal cooling for particles in the wind */
 
     if(All.QuickLymanAlphaProbability > 0) {
-        double dloga = get_dloga_for_bin(P[i].TimeBin);
+        const double dloga = get_dloga_for_bin(P[i].TimeBin);
+        /*Entropy at kick time after this timestep: note this uses this timestep's DtEntropy!*/
+        const double KickEntropy = SPHP(i).Entropy + SPHP(i).DtEntropy * dloga/2;
```

Comment:
You mean predicting Entropy at kick time of the current drift time.

Reply:
Yes, indeed.

Reply:
Problem: I am still confused by the 1/2. 

After a drift the position of particles are t_d. the velocity are t_k = t_d - 1/2. now we compute the force at t_d, then we use force to kick velocity to t_k = t_d. then we recompute step size `alpha`, kick velocity to t_k + `alpha` * 0.5. next we will drift particles to t_k + `alpha`.

We assert Entropy is synced with position. Therefore, drift uses DtEntropy to update Entropy. force calculation requires the entropy variable predicted in the future. However I think we shall use t_d + 1, not t_d + 1/2. For two reasons:
1. notice a few lines later the 'modulated' DtEntropy is divided by dloga, not 0.5 dloga.
2. the BH feedback injected energy shall be deposited in the duration of a full time step, not a half time step (which would increase the BH heating rate, messing up the parameter calibration).

Reply:
In short I argue that the Entropy equation doesn't go with leap-frog. It is an implicit-euler, so there is no 0.5.

Reply:
Yes, you are correct. Well noticed!

Reply:
Fix it and we merge the last three PRs to deliver a 'good' version for #171?